src/model/event.rs

```diff
@@ -1212,6 +1218,26 @@ impl CacheUpdate for ReadyEvent {
             }
         }
 
+        // We may be removed from some guilds between disconnect and ready, so we should handle that.
+        let mut guilds_to_remove = vec![];
+        let ready_guilds_hashset =
+            HashSet::<GuildId>::from_iter(self.ready.guilds.iter().map(|status| status.id()));
+        let shard_data = self.ready.shard.unwrap_or_else(|| [1, 1]);
```

Comment:
`*_else` variants on `Option` and `Result` are preferable when the default value is expensive to compute. In this case, however, `unwrap_or_else` can be substituted with `unwrap_or` since the value is a constant/isn't expensive to compute.
```suggestion
        let shard_data = self.ready.shard.unwrap_or([1, 1]);
```